File: c++/1280.cc
#include <iostream>
#include <cmath>
#include <vector>
#include <algorithm>
#include <iostream>
#include <bits/stdc++.h>

using namespace std;

int cs, degree, step, steps;
double tmp, xmin, xmax, org_xmin, summa, part_sum, r, a;

int main() 
{
	ifstream cin("1280.txt");
	cs = 0;

	while (cin >> degree)
	{
		cs += 1;
		summa = 0;
		steps = 0;
		part_sum = 0;
		vector<double> coeff;
		coeff.clear();
		coeff.resize(0);

		for (int i{0}; i<=degree; ++i)
		{
			cin >> tmp;
			coeff.push_back(tmp);
		}

		vector<double> integ;
		for (int i{0}; i<=degree; ++i)
		{
			integ.push_back(coeff[i]/(i+1));
			cout << integ[i] << endl;
		}

		cin >> xmin >> xmax >> step;
		org_xmin = xmin;

		vector<double> step_vec;
		step_vec.clear();
		step_vec.resize(0);

		r = 0;
		for (int i{0}; i<=degree; ++i)
		{ 
			r += integ[i] * pow(xmin, i+1);
		}
		
		summa -= M_PI * r * r;

		r = 0;
		for (int i{0}; i<=degree; ++i)
		{ 
			r += integ[i] * pow(xmax, i+1);
		}

		summa += M_PI * r * r;

		cout << "Case " << cs << ": " << fixed << setprecision(2) << summa << endl;
		if (step_vec.size() == 0)
		{
			cout << "insufficient volume" << endl;
		}
		else
		{
			for (int i{0}; i<step_vec.size(); ++i)
			{
				if (i > 0) {cout << " ";}
				cout << fixed << setprecision(2) << step_vec[i];
			}
			cout << endl;
		}




	}

}

File: python/1280.py
import sys, math

def main():
	#sys.stdin = open('1280.txt', 'r')

	case = 0
	while True:
		case += 1
		init = sys.stdin.readline()

		if init == "":
			break

		degree = int(init)
		coeff = sys.stdin.readline().split()
		for i in range(degree+1):
			coeff[i] = float(coeff[i])

		init = sys.stdin.readline().split()

		xmin = float(init[0])
		org_min = xmin
		xmax = float(init[1])
		step = int(init[2])

		summa = 0
		part_sum = 0
		steps = 0
		step_arr = []
		while xmin < xmax:
			r = coeff[0]
			for i in range(1,degree+1):
				r += coeff[i] * (xmin ** i)
			a = math.pi * (r ** 2) * 0.00004
			summa += a
			if steps < 8:
				part_sum += a
				if part_sum >= step:
					steps += 1
					step_arr.append(xmin-org_min)
					part_sum -= step
			xmin += 0.00004
			


		print("Case {0}: {1:.2f}".format(case,summa))
		if len(step_arr) == 0:
			print("insufficient volume")
		else:
			s = ""
			for i in range(len(step_arr)):
				if i > 0:
					s += " "
				s += "{0:.2f}".format(step_arr[i])
			print(s)

main()